env.h

```diff
@@ -16,21 +16,30 @@ struct pool {
 
 void envinit(void);
 
+/* create a new environment, with an optional parent */
 struct environment *mkenv(struct environment *);
+/* search environment and its parents for a variable, return the value or NULL if not found */
 struct string *envvar(struct environment *, char *);
+/* add to an environment a variable and its value, replacing the old value there is one */
 void envaddvar(struct environment *, char *, struct string *);
+/* using an environment and its parents variables, evaluate an unevaluated string, returning the result */
 struct string *enveval(struct environment *, struct evalstring *);
+/* search environment and its parents for a rule, return rule or NULL if not found */
 struct rule *envrule(struct environment *, char *);
+/* add rule to an environmental, fails if the rule already exists */
 void envaddrule(struct environment *, struct rule *);
 
+/* create a new rule with the given name, bindings not allocated */
 struct rule *mkrule(char *);
+/* add to a rule a variable and its value */
 void ruleaddvar(struct rule *, char *, struct evalstring *);
 
 /* create a new pool with the given name */
 struct pool *mkpool(char *);
 /* lookup a pool by name. fails if it does not exist */
 struct pool *poolget(char *);
 
+/* return an edge's variable evaluated, optionally shell-escaped */
```

Comment:
These are all fine (thanks!), but could you move these comment changes to a separate commit?

Reply:
Ok, done!